src/v/cluster/rm_stm.cc

```diff
@@ -373,6 +355,33 @@ ss::future<checked<model::term_id, tx_errc>> rm_stm::do_begin_tx(
         co_return tx_errc::fenced;
     }
 
+    std::optional<model::record_batch> batch;
```

Comment:
no need for optional, either make it a function or use `?` operator

Reply:
fixed